### PR TITLE
refactor: move `object` utils from `x-components` to `x-utils`

### PR DESCRIPTION
**Pull request by guillei10**
[EX-5793](https://searchbroker.atlassian.net/browse/EX-5793)

### CODE REVIEW

File: packages/x-components/src/plugins/x-plugin.ts [hunk 1]
```diff
@@ -1,12 +1,13 @@
...
 import { deepMerge } from '@empathyco/x-deep-merge';
+import { forEach } from '@empathyco/x-utils';
 import { PluginObject, VueConstructor } from 'vue';
 import Vuex, { Module, Store } from 'vuex';
 import { FILTERS_REGISTRY } from '../filters/filters.registry';
 import { AnyXStoreModule, RootXStoreState } from '../store/store.types';
 import { cleanGettersProxyCache } from '../store/utils/getters-proxy.utils';
 import { RootXStoreModule } from '../store/x.module';
-import { Dictionary, forEach } from '../utils';
+import { Dictionary } from '../utils';
```
Comment (tiborux) [inline packages/x-components/src/plugins/x-plugin.ts:10]: this type can be removed from x-components, its already in x-utils';

File: packages/x-components/src/plugins/x-plugin.ts [hunk 2]
```diff
@@ -1,12 +1,13 @@
 import { SearchAdapter } from '@empathyco/x-adapter';
 import { deepMerge } from '@empathyco/x-deep-merge';
+import { forEach } from '@empathyco/x-utils';
```
Comment (tiborux) [inline packages/x-components/src/plugins/x-plugin.ts:3]:
```suggestion
import { forEach, Dictionary } from '@empathyco/x-utils';
```

File: packages/x-components/src/x-modules/extra-params/components/snippet-config-extra-params.vue [hunk 3]
```diff
@@ -3,10 +3,11 @@
 </template>
 
 <script lang="ts">
+  import { forEach } from '@empathyco/x-utils';
   import Vue from 'vue';
   import { Component, Watch, Inject, Prop } from 'vue-property-decorator';
   import { xComponentMixin } from '../../../components';
-  import { Dictionary, forEach } from '../../../utils';
+  import { Dictionary } from '../../../utils';
```
Comment (tiborux) [inline packages/x-components/src/x-modules/extra-params/components/snippet-config-extra-params.vue:10]: Same here

File: packages/x-components/src/x-modules/extra-params/components/snippet-config-extra-params.vue [hunk 4]
```diff
@@ -3,10 +3,11 @@
 </template>
 
 <script lang="ts">
+  import { forEach } from '@empathyco/x-utils';
```
Comment (tiborux) [inline packages/x-components/src/x-modules/extra-params/components/snippet-config-extra-params.vue:6]:
```suggestion
  import { forEach, Dictionary } from '@empathyco/x-utils';
```

**Review by LuisMartinez15 — APPROVED** [review]

**Review by tiborux — APPROVED** [review]

**Pull request merged**